src/serialization/base64.hpp

```diff
@@ -0,0 +1,8 @@
+#pragma once
+
+#include "serialization/string_view.hpp"
+
+namespace base64 {
+	std::string decode(utils::string_view encoded);
+	std::string encode(utils::u8string_view bytes);
```

Comment:
I disagree with the decision to represent raw binary data as strings. Binary data should be represented as something like `std::vector<uint8_t>`.

Reply:
this is string_view though not string, and it is probably just used becasue there is no array_view and string_view  can also be used with arrays/vectors.

Reply:
I find `basic_string_view` to be a great "view of continuous container" tool, though you have to be aware it does not own the memory.

I've changed the `std::string` return value to `std::vector<uint8_t>`. The argument to encode is a `basic_string_view<uint8_t>` (to prevent character signedness issues).